src/module/rules/rule-element/battle-form/rule-element.ts

```diff
@@ -57,7 +58,19 @@ class BattleFormRuleElement extends RuleElementPF2e<BattleFormRuleSchema> {
                         { required: false, initial: undefined }
                     ),
                     tempHP: new ResolvableValueField({ required: false, nullable: true, initial: null }),
-                    senses: new fields.ObjectField({ required: false, initial: undefined }),
+                    senses: new RecordField(
+                        new fields.StringField({ required: true, blank: false }),
```

Comment:
okok but then give this a restricted set of choices: the resulting object should then have an index signature with optional keys

Reply:
So no error on choices validation and extract the choices keys as optional keys for the resulting object?

Reply:
I think I've got it working now. If the `StringField` props are a specific type those keys are in the resulting object index in addition to allowing any `string` keys.